File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/941. Valid Mountain Array/C++ solution/faster and better solution.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
       int n = arr.size();
       int i = 0;
       // walk up
       while (i + 1 < n && arr[i] < arr[i + 1])
           i++;
        // peak cannot be the first nor the last element
        if (i == 0 || i == n - 1)
           return false;

        // walk down
        while (i + 1 < n && arr[i] > arr[i + 1])
            i++;

        return i == n - 1;
    }
};

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/941. Valid Mountain Array/Python solution/faster and better solution.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        n = len(arr)
        i = 0
        # wlak up
        while i + 1 < n and arr[i] < arr[i + 1]:
            i += 1
        
        # peak cannot be the first or the last
        if i == 0 or i == n - 1:
            return False
        
        # walk down
        while i + 1 < n and arr[i] > arr[i + 1]:
            i += 1
        
        return i == n - 1